lamkey_nick_ndvi.py

```diff
@@ -0,0 +1,723 @@
+# * Write loop that uses open_clean_bands functions
+# * Use the mask_crop_ndvi function
+# * Calculate the mean ndvi
+# 
+# ## Process multiple scenes for one site
+# Steps required to process all Landsat data
+# * Get list of all directories for the Landsat scenes for the site
+# * For each directory use the steps above for one scene to calculate NDVI for the data in that directory
+# * Loop through that list
+#   * Get bands 4 and 5 (red and NIR)
+#   * Get pixel QA bands
+# * Write loop that uses open_clean_bands functions
+#    * Store bands in a list
+#    * Use that list for the mask_crop_ndvi function
+#    * Store NDVI in a list 
+#    * Calculate mean NDVI
+# * Grab the date associated with each NDVI calculation 
+# * Grab site name for each scene
+# * Save NDVI values of the data for that directory to a list or dataframe that contains average NDVI for each scene at each site
+# 
+# ## Process multiple scenes at multiple sites
+# * Get list of all directories associated with each site name
+# * Write a loop that iterates through multiple site folders
+# * Use workflow from processing multiple scenes to iterate through each file within above loop 
+# * Capture results: Mean NDVI, date, site name to a list or dataframe 
+# * Export dataframe with above values to a csv
+# 
+# ## Create the plot
+
+# ## Things to keep in mind for efficient code
+# * Don't modify the raw data; create an outputs directory
+# * Create functions; 1-3 inputs and 1-2 outputs; consider wrapping function that combines multiple functions
+# * Add checks and tests
+#   * correct types
+#   * assumptions of code
+#   * Identify points of failure
+# 
+#   
+
+# In[1]:
+
+
+# Autograding imports - do not modify this cell
+import matplotcheck.autograde as ag
+import matplotcheck.notebook as nb
+import matplotcheck.timeseries as ts
+from datetime import datetime
+
+
+# In[2]:
+
+
+# Import needed packages in PEP 8 order
+# and no unused imports listed (10 points total)
+
+import os
+from glob import glob
+import warnings
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import geopandas as gpd
+import seaborn as sns
+import rioxarray as rxr
+import xarray as xr
+import earthpy as et
+import matplotlib.dates as mdates
+from matplotlib.dates import DateFormatter
+import pyproj
+
+# Adjust font size and style of all plots in notebook with seaborn
+sns.set(font_scale=1.5, style="whitegrid")
+
+# Apply pyproj to speed up processing
+pyproj.set_use_global_context(True)
+
+# Get data and set the working directory
+data = et.data.get_data('ndvi-automation')
+working_dir = os.path.join(et.io.HOME,
+                           "earth-analytics",
+                           "data")
+if os.path.exists(working_dir):
+    print("Path exists. Working directory set")
+    os.chdir(working_dir)
+else:
+    print("Path does not exist. Creating and setting working directory")
+    os.mkdir(working_dir)
+    os.chdir(working_dir)
+
+# Create outputs directory
```

Comment:
@nlamkey  nice job with the conditional checking of the working directory. I always forget to do this and need to make it a standard practice.